build.gradle.kts

```diff
@@ -5,12 +5,11 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 plugins {
     `kotlin-dsl`
```

Comment:
Technically speaking, you probably want replacing `kotlin-dsl` with `id("...")` so Gradle 8 improved `plugins interpreter` could kick-in here

>Gradle 8.0 introduces an interpreter for the declarative plugins {} blocks in .gradle.kts scripts that make the overall build time around 20% faster. Calling the Kotlin compiler for declarative plugins {} blocks is avoided

Reply:
Nice, except the version must be specified in that case.

```
Plugin [id: 'org.gradle.kotlin.kotlin-dsl'] was not found in any of the following sources:
Plugin [id: 'kotlin.kotlin-dsl'] was not found in any of the following sources:
Plugin [id: 'kotlin-dsl'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (plugin dependency must include a version number for this source)
```

We could hard-code 4.0.6, which is generated as 
```kotlin
val PluginDependenciesSpec.`kotlin-dsl`: PluginDependencySpec
    get() = id("org.gradle.kotlin.kotlin-dsl") version "4.0.6"
```
from https://github.com/gradle/gradle/blob/v8.0.0/build-logic/kotlin-dsl/src/main/kotlin/gradlebuild/kotlindsl/generator/tasks/GenerateKotlinDependencyExtensions.kt

See also
https://github.com/gradle/gradle/blob/v8.0.0/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt#L56-L65

Reply:
They did it somehow here:
https://github.com/gradle/gradle/blob/v8.0.0/build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.build-logic.kotlin-dsl-gradle-plugin.gradle.kts

Reply:
According to https://docs.gradle.org/8.0/userguide/plugins.html#sec:constrained_syntax, `kotlin-dsl` should be accepted special case...

Reply:
indeed, it should be a special case